lxd/instance/drivers/driver_qemu.go

```diff
@@ -2865,17 +2865,35 @@ if [ ! -d "/run/systemd/system/" ]; then
     exit 1
 fi
 
+LIB_SYSTEMD="/lib/systemd"
+[ -d "${LIB_SYSTEMD}/system/" ] || LIB_SYSTEMD="/usr${LIB_SYSTEMD}"
+if [ ! -d "${LIB_SYSTEMD}" ]; then
+    echo "Could not find path to systemd"
+    exit 1
+fi
+
+LIB_UDEV="/lib/udev"
+[ -d "${LIB_UDEV}/rules.d/" ] || LIB_UDEV="/usr${LIB_UDEV}"
+if [ ! -d "${LIB_UDEV}" ]; then
+    echo "Could not find path to udev"
+    exit 1
+fi
+
```

Comment:
I found this a little hard to read. It's more verbose but an alternative might be:
```bash
LIB_SYSTEMD=""
for path in "/lib/systemd/system/" "/usr/lib/systemd"; do
    if [ -d "${path}" ]; then
        LIB_SYSTEMD="${path}"
    fi
done

if [ -z "${LIB_SYSTEMD}" ]; then
    echo "Could not find path to systemd"
    exit 1
fi
```

With that said, you mention that Tumbleweed has symlinks to the expected location. Should we just symlink here as well?

Reply:
The thing is I'm checking for the presence of a child dir before concluding the parent dir is the right one. For the systemd case, I check for `${...}/system`'s existence before retaining the `${...}` part. I had started with a `for` loop be felt it was too verbose. I'll take another shot.

Re Tumbleweed having a compat symlink for `/lib -> /usr/lib`, that's a decision for a distro to make as the effects are far reaching.

Reply:
@markylaing this is what I can come up with:
```
for path in "/lib/systemd" "/usr/lib/systemd"; do
    [ -d "${path}/system" ] || continue
    LIB_SYSTEMD="${path}"
    break
done

if [ ! -d "${LIB_SYSTEMD:-}" ]; then
    echo "Could not find path to systemd"
    exit 1
fi
```
Tell me if you think that reads better.

Reply:
yep looks clearer to me too